### PR TITLE
[W5.11][F12-A4]Chuang Cheng Heng

### PR DESCRIPTION
**Pull request by KevinChuangCH**
Added new command: FindTagCommand, which allows finding persons by their tags.
Ready for review.

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -66,6 +66,19 @@ Examples:
 * `find Betsy Tim John`<br>
   Returns Any person having names `Betsy`, `Tim`, or `John`
 
+### Finding all persons containing any keyword in their tags: `findtag`
```
Comment (jonleeyz) [inline doc/UserGuide.md:69]: Clear example in the User Guide, consistent with the other existing entries. Good!

File: src/seedu/addressbook/commands/FindTagCommand.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+     * Retrieves all persons in the address book whose tags contain some of the specified keywords.
+     *
+     * @param keyTags for searching
+     * @return list of persons found
+     */
+    private List<ReadOnlyPerson> getPersonsWithTagContainingAnyKeyword(Set<Tag> keyTags) {
+        final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
+        for (ReadOnlyPerson person : addressBook.getAllPersons()) {
+            final Set<Tag> tagsOfPerson = person.getTags().toSet();
+            if (!Collections.disjoint(tagsOfPerson, keyTags)) {
```
Comment (jonleeyz) [inline src/seedu/addressbook/commands/FindTagCommand.java:63]: Great use of Collections.disjoint() method here.

File: src/seedu/addressbook/commands/FindTagCommand.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package seedu.addressbook.commands;
```
Comment (jonleeyz) [inline src/seedu/addressbook/commands/FindTagCommand.java:1]: Clean, neat and self explanatory code!

File: test/input.txt [hunk 4]
```diff
@@ -106,6 +106,26 @@
   # find multiple with some keywords
   find Charlie Betsy
 
+##########################################################
+# test find persons by tag command
+##########################################################
+
+  # should consider no keywords as invalid command format
```
Comment (jonleeyz) [inline test/input.txt:113]: Nice work encompassing most, if not all, of the possible test cases.

File: src/seedu/addressbook/commands/FindTagCommand.java (hunk 2 above)
Comment (jonleeyz) [reply src/seedu/addressbook/commands/FindTagCommand.java:63]: Could the checking for any similar tags be abstracted out into another helper method?

**Review by OuyangDanwen — COMMENTED** [review]
Great effort in writing neat code. Kudos for adding the Junit tests. Keep it up! You can close the PR. @KevinChuangCH

**Pull request closed without merging**